core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/replace.kt

```diff
@@ -13,4 +15,61 @@ class ReplaceTests {
         conv.columnNames() shouldBe listOf("b")
         conv.columnTypes() shouldBe listOf(typeOf<Double>())
     }
+
+    @Test
+    fun `unfold primitive`() {
+        val a by columnOf("123")
+        val df = dataFrameOf(a)
+
+        val conv = df.replace { a }.unfold {
+            "b" from { it }
+            "c" from { DataRow.readJsonStr("""{"prop": 1}""") }
+        }
+
+        val b = conv["a"]["b"]
+        b.type() shouldBe typeOf<String>()
+        b.values() shouldBe listOf("123")
+
+        val c = conv["a"]["c"]["prop"]
+        c.type() shouldBe typeOf<Int>()
+        c.values() shouldBe listOf(1)
+    }
+
+    @Test
+    fun `unfold properties`() {
+        val col by columnOf(A("1", 123, B(3.0)))
```

Comment:
Is this case "More fine-grained toDataFrame. Instead of converting 20-30 properties to 2-3 level of nesting all at once user can choose to convert toDataFrame(maxDepth = 0) and unfold required properties to whatever level they need" covered here, in this test?

Reply:
Technically, yes. I intend to have a more representative example as a part of compiler plugin demo. There's a tree of objects with many properties and potentially deep nesting from konsist library. It will be a good illustration. But here it merely unfolds one specific column up to 2 levels.